hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -1073,10 +1074,30 @@ int QuectelNcpClient::getRuntimeBaudrate() {
         runtimeBaudrate = QUECTEL_NCP_RUNTIME_SERIAL_BAUDRATE_BG95_M5;
     } else if (ncpId() == PLATFORM_NCP_QUECTEL_EG91_NAX) {
         runtimeBaudrate = QUECTEL_NCP_RUNTIME_SERIAL_BAUDRATE_EG91_NAX;
+    } else if (ncpId() == PLATFORM_NCP_QUECTEL_EG91_E || ncpId() == PLATFORM_NCP_QUECTEL_EG91_EX) {
+        if (fwVersion_ >= 8) {
+            runtimeBaudrate = QUECTEL_NCP_RUNTIME_SERIAL_BAUDRATE_EG91_EX;
+        }
     }
     return runtimeBaudrate;
 }
 
+int QuectelNcpClient::getAppFirmwareVersion() {
+    // example output: EG91NAXGAR07A03M1G_30.004.30.004
+    // Right now we only care about EG91EFBR06AxxM4G*
+    auto resp = parser_.sendCommand("AT+QGMR");
+    int ver = 0;
+    int major = 0;
+    int n = CHECK_PARSER(resp.scanf("EG91EFBR06A%dM4G", &major));
```

Comment:
It looks like this firmware version `EG91NAXGAR07A03M1G_30.004.30.004` is specific to the EG91-NAX? 

My EG91EX has this firmware version:
```
0000012547 [ncp.at] TRACE: > AT+QGMR
0000012552 [ncp.at] TRACE: < EG91EXGAR08A14M1G_20.200.20.200
0000012556 [ncp.at] TRACE: < OK
0000012558 [ncp.client] INFO: QUERIED FW VERSION 0
```

So we will query for both EG91-E and EG91-EX modems, but the only firmware string that should match will be the EG91-E version. Is there an analogous EG91-EX firmware version we want to support 921600 baud on?

Reply:
Ahh, good catch.  I was pretty sure I had tried this with an EX modem.  What SKU is that with the EX modem?  I'll make sure it also has the same requirement.

Reply:
M524

Reply:
I flashed the oldest firmware available for `EG91EXGAR08A03M1G_01.003.01.003` and it supports 921k baud, so we can limit the version check for EG91-E only, and force EG91-EX to 921k.  I updated the PR to reflect this.